src/EventsService/EventsService.Application/UseCases/Dates/Commands/DeleteDate/DeleteDateHandler.cs

```diff
@@ -0,0 +1,26 @@
+namespace EventsService.Application.UseCases.Dates.Commands.DeleteDate;
+
+using EventsService.Application.Common.Extensions;
+using EventsService.Domain.Contracts;
+using MediatR;
+
+public class DeleteDateHandler : IRequestHandler<DeleteDateCommand>
+{
+    private readonly IDateRepository _dateRepository;
+
+    public DeleteDateHandler(IDateRepository dateRepository)
+    {
+        this._dateRepository = dateRepository;
+    }
+
+    public async Task Handle(DeleteDateCommand request, CancellationToken cancellationToken)
+    {
+        var date = await this._dateRepository.GetByIdAsync(request.Id, cancellationToken);
```

Comment:
In order to reduce the amount of code, you can use the ?? operator:
```suggestion
        var date = await this._dateRepository.GetByIdAsync(request.Id, cancellationToken)
            ?? throw new EntityNotFoundException(nameof(date), request.Id);
```